crates/solver-types/src/utils/eip712.rs

```diff
@@ -81,6 +82,332 @@ impl Eip712AbiEncoder {
 	}
 }
 
+/// Recovers the user address from a Permit2 signature using ecrecover.
+///
+/// # Arguments
+///
+/// * `digest` - The EIP-712 digest that was signed (32 bytes)
+/// * `signature` - The signature string with format "0x00<65_byte_signature>"
+///
+/// # Returns
+///
+/// Returns the recovered user address or an error if recovery fails.
+pub fn ecrecover_user_from_signature(
```

Comment:
Pretty sure there are libraries for this, what's the benefit of attempting to do this manually?